src/lib.rs

```diff
@@ -680,7 +646,9 @@ where
                 input_str.into()
             }
         } else {
-            // we cannot handle `~otheruser/` paths yet
+            // `FnOnce() -> Option<P>` api does not allow for handling `~otheruser/` paths here.
+            // Would need to make breaking change to `FnOnce(username: Option<&str>) -> Option<P>`
+            // in order to handle it.
```

Comment:
As I said, breaking change is okay here, so let's change `HD` to `FnOnce(Option<&str>) -> Option<P>`, and propagate this API upwards to avoid code duplication.

Reply:
Yes - That's what I'll do. Just haven't pushed a commit for it yet as I've been working on getting Windows working. The only commits so far are just merging in the changes you made to the code after I posted the initial PR; so everything is in an intermediate / not ready for review state. 

I'm going to push several commits today (including Windows, which works, I just need to clean it up); so let's wait on reviewing the code until all that is complete. I'll leave a message on the main thread when I think it's ready for review. Thanks!